diff --git a/packages/kokkos/core/src/Kokkos_Macros.hpp b/packages/kokkos/core/src/Kokkos_Macros.hpp
index c221c2f..de04ef0 100644
--- a/packages/kokkos/core/src/Kokkos_Macros.hpp
+++ b/packages/kokkos/core/src/Kokkos_Macros.hpp
@@ -253,7 +253,7 @@
     #endif
   #endif
   #if ( 1200 <= KOKKOS_COMPILER_INTEL ) && ! defined( KOKKOS_ENABLE_ASM ) && ! defined( _WIN32 )
-    #define KOKKOS_ENABLE_ASM 1
+    #define KOKKOS_ENABLE_ASM 0
   #endif

   #if ( 1200 <= KOKKOS_COMPILER_INTEL ) && ! defined( KOKKOS_FORCEINLINE_FUNCTION )
@@ -330,8 +330,11 @@
           defined(__POWERPC__) || \
           defined(__ppc__) || \
           defined(__ppc64__) || \
-          defined(__PGIC__) )
-    #define KOKKOS_ENABLE_ASM 1
+          defined(__PGIC__) || \
+          defined(__AArch64__) || \
+          defined(__aarch64__)  \
+          )
+    #define KOKKOS_ENABLE_ASM 0
   #endif

 #endif
diff --git a/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp b/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp
index abd845d..6ce7091 100644
--- a/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp
+++ b/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp
@@ -46,6 +46,7 @@

 /*--------------------------------------------------------------------------*/

+#ifdef CEG
 #if ( KOKKOS_ENABLE_ASM )
   #if defined( __arm__ )
     /* No-operation instruction to idle the thread. */
@@ -64,6 +65,12 @@
   #define YIELD  sched_yield()
 #endif

+/* CEG bit */
+#else
+  #include <sched.h>
+  #define YIELD  sched_yield()
+#endif
+
 /*--------------------------------------------------------------------------*/

 namespace Kokkos {
