diff -ruN Trilinos-trilinos-release-12-4-2.old/packages/kokkos/core/src/Kokkos_Macros.hpp Trilinos-trilinos-release-12-4-2/packages/kokkos/core/src/Kokkos_Macros.hpp
--- Trilinos-trilinos-release-12-4-2.old/packages/kokkos/core/src/Kokkos_Macros.hpp	2015-11-12 15:57:46.000000000 +0000
+++ Trilinos-trilinos-release-12-4-2/packages/kokkos/core/src/Kokkos_Macros.hpp	2016-09-09 23:46:32.342977179 +0100
@@ -253,7 +253,7 @@
     #endif
   #endif
   #if ( 1200 <= KOKKOS_COMPILER_INTEL ) && ! defined( KOKKOS_ENABLE_ASM ) && ! defined( _WIN32 )
-    #define KOKKOS_ENABLE_ASM 1
+    #define KOKKOS_ENABLE_ASM 0
   #endif
 
   #if ( 1200 <= KOKKOS_COMPILER_INTEL ) && ! defined( KOKKOS_FORCEINLINE_FUNCTION )
@@ -330,8 +330,11 @@
           defined(__POWERPC__) || \
           defined(__ppc__) || \
           defined(__ppc64__) || \
-          defined(__PGIC__) )
-    #define KOKKOS_ENABLE_ASM 1
+          defined(__PGIC__) || \
+          defined(__AArch64__) || \
+          defined(__aarch64__)  \
+          )
+    #define KOKKOS_ENABLE_ASM 0
   #endif
 
 #endif
diff -ruN Trilinos-trilinos-release-12-4-2.old/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp Trilinos-trilinos-release-12-4-2/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp
--- Trilinos-trilinos-release-12-4-2.old/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp	2015-11-12 15:57:46.000000000 +0000
+++ Trilinos-trilinos-release-12-4-2/packages/kokkos/core/src/impl/Kokkos_spinwait.cpp	2016-09-09 23:46:32.342977179 +0100
@@ -46,6 +46,7 @@
 
 /*--------------------------------------------------------------------------*/
 
+#ifdef CEG
 #if ( KOKKOS_ENABLE_ASM )
   #if defined( __arm__ ) || defined( __aarch64__ )
     /* No-operation instruction to idle the thread. */
@@ -62,6 +63,12 @@
 #else
   #include <sched.h>
   #define YIELD  sched_yield()
+#endif
+
+/* CEG bit */
+#else
+  #include <sched.h>
+  #define YIELD  sched_yield()
 #endif
 
 /*--------------------------------------------------------------------------*/
