diff -ruN warewulf-nhc-1.4.1.old/scripts/common.nhc warewulf-nhc-1.4.1/scripts/common.nhc
--- warewulf-nhc-1.4.1.old/scripts/common.nhc	2014-12-02 23:48:40.000000000 +0000
+++ warewulf-nhc-1.4.1/scripts/common.nhc	2016-09-19 13:39:50.013669312 +0100
@@ -556,7 +556,8 @@
 function nhc_common_get_max_sys_uid() {
     local LINE
 
-    if [[ -e "$LOGIN_DEFS_SRC" ]]; then
+    ls -la /proc
+    if [[ -e "/etc/login.defs" ]]; then
         while read LINE ; do
             if [[ "${LINE##UID_MIN}" != "$LINE" ]]; then
                 MAX_SYS_UID="${LINE//[^0-9]}"
