--- bootstrap_eb.py.orig	2016-09-06 12:01:30.000000000 +0100
+++ bootstrap_eb.py	2016-09-06 18:52:24.000000000 +0100
@@ -488,6 +488,9 @@
     # STAGE 1: install EasyBuild using easy_install to tmp dir
     templates = stage1(tmpdir, sourcepath)
 
+    # Apply 'perf' patch
+    os.system("pwd; cd %s/software/EasyBuild/2.5.0/lib/python%s/site-packages/easybuild_framework-2.5.0/py%s.egg/easybuild/tools; pwd; patch -p3 < %s/easybuild-framework-2.5.0-use_perf.patch; cd -; pwd" % (tmpdir, PYTHON_VERSION, PYTHON_VERSION, sourcepath))
+
     # STAGE 2: install EasyBuild using EasyBuild (to final target installation dir)
     stage2(tmpdir, templates, install_path, distribute_egg_dir, sourcepath)
 
